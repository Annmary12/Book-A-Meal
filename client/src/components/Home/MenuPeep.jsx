import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DisplayMeals from './DisplayMeals.jsx';
import LoadingMeals from './LoadingMeals.jsx';
import styles from './menupeep.scss';

/**
 * @class MenuPeep
 */
class MenuPeep extends Component {
  /**
   * @returns {undefined} - undefined
   */
  componentDidMount() {
    this.props.getMenus();
  }

  /**
   * @returns {JSX} - React JSX
   */
  render() {
    return (
      <div className={styles.peep}>
        <div className={styles.heading}>
          <span>~~ <b>Peep into today&#39;s menus</b> ~~</span>
        </div>
        { this.props.loading ? <LoadingMeals /> : <DisplayMeals meals={this.props.meals} /> }
      </div>
    );
  }
}


MenuPeep.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  getMenus: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired
};

export default MenuPeep;
