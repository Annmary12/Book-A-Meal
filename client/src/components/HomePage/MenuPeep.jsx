import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DisplayMeals from './DisplayMeals.jsx';
import LoadingMeals from './LoadingMeals.jsx';

const MenuPeep = ({loading, meals}) => (
  <div className="content">   
    <div className="menu-peak">
      <span>~~  </span>
      <span>
        <b>Peep into today's menus</b>
      </span>
      <span>  ~~</span>
    </div>
    { loading ? <LoadingMeals /> : <DisplayMeals meals={meals} /> }
  </div>
)

MenuPeep.propTypes = {
  meals: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired
}

export default MenuPeep;