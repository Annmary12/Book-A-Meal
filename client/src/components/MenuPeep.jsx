import React, { Component } from 'react';
import MealCard from './MealCard.jsx';
import FriedRiceImg from '../assets/img/fried-rice.jpg';

class MenuPeep extends Component{
  render() {
    return (
      <div className="content">   
        <div className="menu-peak">
          <span>~~  </span><span><b>Peep into today's menu</b></span><span>  ~~</span>
        </div>
        <div className="meals">
          <MealCard image={FriedRiceImg} description="delicious meal" title="fried-rice" price="3000" />
        </div>
      </div>
    )
  }
}

export default MenuPeep;