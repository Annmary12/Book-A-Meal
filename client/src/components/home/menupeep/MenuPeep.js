import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DisplayMeals from '../../DisplayMeals/DisplayMeals.js';
import Loader from '../../Loader.js';
import styles from './menu-peep.scss';

/**
 * @class MenuPeep
 */
class MenuPeep extends Component {
  /**
   * @returns {undefined} - undefined
   */
  componentDidMount() {
    this.props.peepMenus();
  }

  /**
   * @returns {JSX} - React JSX
   */
  render() {
    const { meals, loggedIn, loading } = this.props;
    return (
      <div className={styles.peep}>
        <div className={styles.heading}>
          <span>~~ <b>Peep into today&#39;s menus</b> ~~</span>
        </div>
        { loading
          ? <Loader />
          : <DisplayMeals loggedIn={loggedIn} meals={meals} /> }
      </div>
    );
  }
}


MenuPeep.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  peepMenus: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  loggedIn: PropTypes.bool.isRequired
};

export default MenuPeep;
