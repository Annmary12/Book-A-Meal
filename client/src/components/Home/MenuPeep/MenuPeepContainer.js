import { connect } from 'react-redux';
import { peepMenus } from '../../../actions/peepMenus.js';
import MenuPeep from './MenuPeep.js';

const mapStateToProps = state => ({
  loading: state.menusPeep.isFetching,
  meals: state.menusPeep.meals,
  loggedIn: state.auth.loggedIn
});

export default connect(mapStateToProps, {
  peepMenus
})(MenuPeep);
