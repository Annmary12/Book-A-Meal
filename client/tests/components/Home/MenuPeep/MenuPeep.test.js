import React from 'react';
import MenuPeep from '../../../../src/components/Home/MenuPeep/MenuPeep';

describe('MenuPeep component', () => {
  const props = {
    meals: [],
    peepMenus: jest.fn(),
    loading: false,
    loggedIn: true
  }
  it('should match snapshot if loading is false', () => {
    const wrapper = shallow(
      <MenuPeep
        {...props}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot if loading is true', () => {
    const wrapper = shallow(<MenuPeep
      {...props}
      loading={true}
    />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should call peepMenu prop function on mount', () => {
    const wrapper = shallow(
      <MenuPeep
        {...props}
      />
    );

    wrapper.instance().componentDidMount();
    expect(props.peepMenus).toBeCalled();
  });
});