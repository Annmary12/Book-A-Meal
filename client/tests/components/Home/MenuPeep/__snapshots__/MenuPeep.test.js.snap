// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuPeep component should match snapshot if loading is false 1`] = `
<div
  className="peep"
>
  <div
    className="heading"
  >
    <span>
      ~~ 
      <b>
        Peep into today's menus
      </b>
       ~~
    </span>
  </div>
  <DisplayMeals
    loggedIn={true}
    meals={Array []}
  />
</div>
`;

exports[`MenuPeep component should match snapshot if loading is true 1`] = `
<div
  className="peep"
>
  <div
    className="heading"
  >
    <span>
      ~~ 
      <b>
        Peep into today's menus
      </b>
       ~~
    </span>
  </div>
  <Loader
    message="loading..."
    width={40}
  />
</div>
`;
